--- origsrc/pxz/pxz.c	2011-03-28 19:28:12.000000000 -0500
+++ src/pxz/pxz.c	2011-03-28 20:32:49.248475400 -0500
@@ -21,7 +21,7 @@
 
 #include <string.h>
 #include <stdio.h>
-#include <stdio_ext.h>
+//#include <stdio_ext.h>
 #include <stdlib.h>
 #include <inttypes.h>
 #include <unistd.h>
@@ -220,7 +219,7 @@ int close_stream( FILE *f ) {
 		return EOF;
 	}
 	
-	if ( fclose(f) && (__fpending(f) || errno != EBADF) ) {
+	if ( fclose(f) && (errno != EBADF) ) {
 		return EOF;
 	}
 	
