--- origsrc/pxz/pxz.c	2011-05-30 16:54:02.720399400 -0500
+++ src/pxz/pxz.c	2011-05-30 16:59:52.736419100 -0500
@@ -25,7 +25,7 @@
 #include <stdlib.h>
 #include <inttypes.h>
 #include <unistd.h>
-#include <error.h>
+#include <stdarg.h>
 #include <errno.h>
 #include <sys/stat.h>
 #include <sys/mman.h>
@@ -119,6 +119,28 @@ const struct option long_opts[] = {
 	{ NULL,             0,                 NULL,   0 }
 };
 
+static void error (int, int, const char *, ...);
+
+static void error (int status, int errnum, const char *fmt, ...)
+{
+  va_list ap;
+
+  fflush (stdout);
+  fprintf (stderr, "%s: ", program_invocation_name);
+
+  va_start (ap, fmt);
+  vfprintf (stderr, fmt, ap);
+  va_end (ap);
+
+  if (errnum != 0)
+    fprintf (stderr, ": %s", strerror (errnum));
+
+  fprintf (stderr, "\n");
+
+  if (status != 0)
+    exit (status);
+}
+
 void __attribute__((noreturn)) run_xz( char **argv ) {
 	execvp(XZ_BINARY, argv);
 	error(0, errno, "execution of "XZ_BINARY" binary failed");
