--- origsrc/pxz/pxz.c	2011-03-28 19:28:12.000000000 -0500
+++ src/pxz/pxz.c	2011-03-28 20:32:49.248475400 -0500
@@ -25,7 +25,7 @@
 #include <stdlib.h>
 #include <inttypes.h>
 #include <unistd.h>
-#include <error.h>
+#include <stdarg.h>
 #include <errno.h>
 #include <sys/stat.h>
 #include <sys/mman.h>
@@ -119,6 +119,22 @@ const struct option long_opts[] = {
 	{ NULL,             0,                 NULL,   0 }
 };
 
+static void error(int, int, const char *, ...);
+
+static void error(int status, int errnum, const char *format, ...) {
+	va_list ap;
+	va_start(ap, format);
+	char msg[1024];
+
+	vsprintf(msg, format, ap);
+	va_end(ap);
+
+	fprintf(stderr, "%s: %s: %s\n", program_invocation_name,
+			msg, strerror(errnum));
+	if (status != 0)
+		exit(status);
+}
+
 void __attribute__((noreturn)) run_xz( char **argv ) {
 	execvp(XZ_BINARY, argv);
 	error(0, errno, "execution of "XZ_BINARY" binary failed");
